nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java

```diff
@@ -118,7 +121,28 @@ protected String computeUserAgent() {
         } catch (IOException e) {
             nuxeoPart += "Unknown";
         }
-        return Version.userAgent() + nuxeoPart;
+        String okhttpPart = runReflection(() -> {
+            // handle okhttp 3 version
+            Class<?> okhttp3VersionClass = Class.forName("okhttp3.internal.Version");
+            Method userAgentMethod = okhttp3VersionClass.getMethod("userAgent");
+            return (String) userAgentMethod.invoke(null);
+        }, () -> runReflection(() -> {
+            // handle okhttp 4 version
+            Class<?> okhttp4UtilClass = Class.forName("okhttp3.internal.Util");
+            Field userAgentField = okhttp4UtilClass.getField("userAgent");
+            return (String) userAgentField.get(null);
+
+        }, () -> "okhttp/Unknown"));
+        return okhttpPart + " " + nuxeoPart;
+    }
+
+    protected String runReflection(ThrowableSupplier<String, ReflectiveOperationException> supplier,
```

Comment:
I would have named this in a way that reflects the error handling aspect ? handleThrowableSupplier maybe ?
To me runReflection reflects what the suppliers are doing. 
That I would have put in a comment before the call line 124 like "Get version through reflection" ?

Reply:
I see your point, indeed the method is not well named. `handleThrowableSupplier` doesn't express the fact that we're running the given supplier, I will think about a better name and propose a change.

Reply:
I was trying to generify the method to give it a generic word but we can't catch exception by parameter type, so we end up to declare `ReflectiveOperationException` as type parameter of `ThrowableSupplier` on the method, which tied the method to the reflection.
I will rename the method `runReflectiveSupplierOrFallback`.